cabal-testsuite/PackageTests/Regression/T6011/cabal.test.hs

```diff
@@ -0,0 +1,4 @@
+import Test.Cabal.Prelude
+-- Test build-tool-depends between two packages
+main = cabalTest $ do
+    cabal "v2-build" ["client", "--disable-per-component"]
```

Comment:
Looking at this, we probably should do something special in `ProjectPlanning` based on whether per-component build is disabled (either by flag or `build-type: Custom`). We **should not** default to do more in all scenarios just in case if per component build is disabled: everyone would pay for a corner-case.

Reply:
Could you elaborate on this? What would the "something special" look like? In what sense is everyone paying for the corner case in the current situation?